src/TimeModeling/judiVector.jl

```diff
@@ -713,19 +713,25 @@ iterate(S::judiVector, state::Integer=1) = state > S.nsrc ? nothing : (S.data[st
 
 function cumsum(x::judiVector;dims=1)
     dims == 1 || dims == 2 || throw(judiVectorException("Dimension $(dims) is out of range for a 2D array"))
-    y = deepcopy(x)
+    y = (dims == 1 ? x.geometry.dt[1] : 1f0) * x        # receiver dimension is non-dimensional
     for i = 1:x.nsrc
-        cumsum!(y.data[i], x.data[i], dims=dims)
+        cumsum!(y.data[i], y.data[i], dims=dims)
```

Comment:
Yu can't do that, that's not in place and will create a temporary array. Most in-place operation do not allow to alias the result to the input (you should check it up), so you can have the result be the same object as the input. so like 716 should only get ` h = dims == 1 ? x.geometry.dt[1] : 1f0` then `y = similar(x)` then `cumsum!(y, x); y .*= h` (`rmul!` even better bit not implemented I think)